kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/predefined/Erase.java

```diff
@@ -43,8 +42,11 @@
 
     public Erase() {
         super();
-        String extraRoles = Gateway.getProperties().getString("PredefinedStep.Erase.roles");
-        getProperties().put(AGENT_ROLE.getName(), "Admin" + (extraRoles != null ? ","+extraRoles : ""));
+
+        if (Gateway.getProperties().getBoolean("PredefinedStep.AgentRole.enableAdmin", false)) {
+            String extraRoles = Gateway.getProperties().getString("PredefinedStep.Erase.roles");
+            getProperties().put(AGENT_ROLE.getName(), ADMIN_ROLE.getName() + (extraRoles != null ? ","+extraRoles : ""));

```

Comment:
Could we replace `extraRoles != null` with `StringUtils.isNotBlank()`?

Reply:
done